python/graphstorm/sagemaker/README.md

```diff
@@ -0,0 +1,198 @@
+To launch the services with a docker compose file, we can use `docker compose -f docker-compose.yaml up`.
+This will launch the container and execute its entry point.
+
+To emulate a SageMaker distributed execution environment based on the image (suppose the docker image is named graphstorm-sagemaker-dev:v1) you built previously you would need a Docker compose file that looks like this:
+```
+version: '3.7'
+
+networks:
+  gfs:
+    name: gfs-network
+
+services:
+  algo-1:
+    image: graphstorm-sagemaker-dev:v1
+    container_name: algo-1
+    hostname: algo-1
+    networks:
+      - gfs
+    command: 'xxx'
+    environment:
+      SM_TRAINING_ENV: '{"hosts": ["algo-1", "algo-2", "algo-3", "algo-4"], "current_host": "algo-1"}'
+      WORLD_SIZE: 4
+      MASTER_ADDR: 'algo-1'
+      AWS_REGION: 'us-west-2'
+    ports:
+      - 22
+    working_dir: '/opt/ml/code/'
+
+  algo-2:
+      [...]
+
+```
+Some explanation on the above elements (see the [official docs](https://docs.docker.com/compose/compose-file/) for more details):
+
+* `image`: Determines which image you will use for the container launched.
+* `environment`: Determines the environment variables that will be set for the container once it launches.
+* `command`: Determines the entrypoint, i.e. the command that will be executed once the container launches.
+
+To help you generate yaml file automatically, we provide a Python script that
+builds the docker compose file for your, `generate_sagemaker_docker_compose.py`.
+Note that the script uses the [PyYAML](https://pypi.org/project/PyYAML/) library.
+
+This file has 4 required arguments that determine the Docker compose file that will be generated:
+
+* `--num-instances`: The number of instances we want to launch.
+This will determine the number of `algo-x` `services` entries our compose file ends up with.
+* `--aws-access-key-id`: The AWS access key ID for accessing S3 data within docker
+* `--aws-secret-access-key`: The AWS secret access key for accessing S3 data within docker.
+* `--aws-session-token`: The AWS session toekn used for accessing S3 data within docker.
+
+The rest of the arguments are passed on to `sagemaker_train.py`
+* `--task-type`: Training task type.
+* `--graph-data-s3`: S3 location of input training graph.
+* `--graph-name`: Name of the input training graph.
+* `--train-yaml-s3`: S3 location of training yaml file.
+* `--custom-script`: Custom training script provided by a customer to run customer training logic. This should be a path to the python script within the docker image.
+
+If you want to pass other arguements to `sagemaker_train.py`, you can simply append those arguments after the above arguments.
+
+The above will create a Docker compose file named `docker-compose-${task-type}-${num-instances}-train.yml`, which we can then use to launch the job with (for example):
+
+```bash
+docker compose -f docker-compose-node-classification-4-train.yml up
```

Comment:
Won't the task be using an underscore here? `node_classification`?

Reply:
You are right